pkgs/development/beam-modules/fetch-rebar-deps.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, rebar3, curl }:
+
+{ name, version, sha256, src
+, meta ? {}
+}:
+
+with stdenv.lib;
+
+stdenv.mkDerivation ({
+  name = "rebar-deps-${name}-${version}";
+
+  phases = [ "downloadPhase" "installPhase" ];
+
+  downloadPhase = ''
+    cp ${src} .
+    HOME='.' DEBUG=1 ${rebar3}/bin/rebar3 get-deps
```

Comment:
For what it's worth, this will only work on `rebar3-open` because it is not hermetic. How does this really differ from the current `fetchHex` (aside from it possibly getting non-hex deps)?

Reply:
The current way of dealing erlang dependencies is not viable. Not only hermetic patch breaks rebar3 for everyone, but it also heavily relies on undocumented rebar3 features and file locations.

With the current approach in order to update a package one needs to

1) Ask @gleber to update hex index cache repo (note that hex index is not actually needed. it's only used to avert rebar3 errors)
2) Wait until index hash repo is merged and the derivation is updated in nixpkgs repo
3) Ping someone to run hex2nix and update hex packages in this repo. 
4) Wait again until the new PR is reviewed and merged
5) Manually convert lock file contents to the Nix expression
6) Pray that nothing doesn't have non-trivial build steps
8) Pray especially hard that no one uses rebar3 `override` feature

With the new approach every package can be maintained basically independently.